### DISCUSSION TITLE
Nonlinear velocity profile

### DISCUSSION THREAD

**Discussion by sxroller** [nextfoam/baram#138] [Q&A]
Dear community,
I'm trying to get Eurocode like wind profile which is nonlinerar. Example of desired wind profile below:
![image](https://github.com/user-attachments/assets/0894bab9-3b4f-473e-b85b-e01396a3286e)
From baramFlow user guide I have concluded that should use velocity inlet spatial distribution with csv file. So I prepared file with velocity connected with z axis. Piece of the file below:
```
0.0, 0.0, 0.0, 17.0, 0.0, 0.0
0.0, 0.0, 1.0, 17.0, 0.0, 0.0
0.0, 0.0, 2.0, 17.0, 0.0, 0.0
0.0, 0.0, 3.0, 18.0, 0.0, 0.0
0.0, 0.0, 4.0, 19.0, 0.0, 0.0
0.0, 0.0, 5.0, 20.0, 0.0, 0.0
0.0, 0.0, 6.0, 20.0, 0.0, 0.0
0.0, 0.0, 7.0, 21.0, 0.0, 0.0
0.0, 0.0, 8.0, 21.0, 0.0, 0.0
0.0, 0.0, 9.0, 22.0, 0.0, 0.0
0.0, 0.0, 10.0, 22.0, 0.0, 0.0
```
And it wont work, after initialisation I recive error:
![image](https://github.com/user-attachments/assets/829bc0e3-4688-40c8-9c1a-eed9c7a1adec)

Please guide me what I am doing wrong and how to get desired wind profile

**Comment by kdg5424** [comment]
Hello @sxroller. This is Donggyu from NEXTFOAM.
It looks like there is no problem with your csv file.
Or Could you try profile with 30 z-axis again?
If the problem happens again, please let me know and upload your case file with csv file.

Best regards

Donggyu Kim

**Comment by sxroller** [comment]
I prepared a new csv file that exactly matches the size of my case (max z = 150 m) and still get an error. 

![image](https://github.com/user-attachments/assets/d12cc5f3-e895-4b63-982a-57c68165ac0b)

I am attaching my csv file. I would like to upload the entire baramFlow folder, but it exceeds the git file limits. I may have made a mistake somewhere else, but with the velocity inlet set as a constant magnitude works fine.
Thank you very much for help
[Wind.csv](https://github.com/user-attachments/files/17561083/Wind.csv)

**Comment by kdg5424** [comment]
Hi. This is Donggyu.
We changed your csv like this : 

![image](https://github.com/user-attachments/assets/e9002ad8-c52c-4a0c-ae63-40653e7c26c1)

![image](https://github.com/user-attachments/assets/d35a9a1c-0fca-4fbd-959b-3f68580c3ce3)

And it works.

Could you change your csv like the picture and try again?

Best regards

Donggyu Kim

**Comment by sxroller** [comment]
Wow, adding y coordiante did the trick. My analysis finally works the way I want.  Thank you so much
![Zrzut ekranu 2024-10-30 181041](https://github.com/user-attachments/assets/cfcc8371-6cb7-43d9-8b68-fcdabc07f7bd)

**Comment by kdg5424** [comment]
I'm very glad to hear that.
In 3D domain, the velocity inlet is defined as a plane.
And it means that velocity profile should be defined for the plane.
So I recommend you to add or implement the trick I told you.

Thank you

Best regards

Donggyu Kim
